### DISCUSSION TITLE
Namespace support

### DISCUSSION THREAD

**Discussion by mr-miles** [openbao/openbao#778] [General]
I'm a long-time vault user and, reading the "project direction & roadmap", was really keen to contribute a PR to get the namespace support going.

I'm very familiar with how the feature should operate.  I wanted to find out:

 - has anyone looked at this before (my search of chats/discussions didn't throw anything up, but that might be user error)
 - are there any other considerations to keep in mind that aren't apparent from the codebase
 - what I should do to fit in with the norms around discussion/transparency for this sort of contribution (since I imagine it could touch a lot of the codebase)

Cheers!

**Comment by cipherboy** [comment]
\o hey @mr-miles -- we've had a working group going for a couple of months now. Getting a few sign-offs and then I think the RFC will be published for adding support and a bit of a feature roadmap. The WG will do the implementation as well most likely, there's two parallel PoCs at the moment.

But, outside of some ideas for the future of the feature, we don't have implementers for that. So I'd suggest holding off for now on designing or implementing, and perhaps see if you can contribute to testing and future enhancements instead if you're willing!

**Comment by mr-miles** [comment]
Hi @cipherboy - yes, happy to help on that!

What's the best place to look for things that you'd like to get worked on near-term in openbao?  I wasn't totally clear on how to combine the  [Project Direction & Roadmap](https://github.com/openbao/openbao/issues/569), the open issues list, and the open PRs to see what's useful but not actively being worked on right now.

**Reply by cipherboy** [reply]
@mr-miles Generally we've assigned people when it's being actively worked on, though sometimes we're a  little inconsistent. In general most things are fair game. Once https://github.com/openbao/go-kms-wrapping/pull/14 lands, we'll need a corresponding change in OpenBao to do this... 

I guess the real question is, what areas interest you? Plugin registry could probably use an owner (@JanMa has gotten busy with other things but is otherwise happy to collaborate), profile system (from our blog post) could use some more thoughts and discussion if you want to go down that route. Or the ACL and auditing systems could also use a fair bit of help :-)

**Reply by cipherboy** [reply]
@mr-miles If you didn't see it, https://github.com/openbao/openbao/issues/787 has the RFC on namespaces btw :-)

**Reply by cipherboy** [reply]
@mr-miles Support for namespaces has landed in `main` and should be available in v2.3.0 Beta release!
